hal/interfaces/test/TestFlashHeterogeneous.cpp

```diff
@@ -0,0 +1,57 @@
+#include "hal/interfaces/FlashHeterogeneous.hpp"
+#include "gtest/gtest.h"
+#include <cstdint>
+
+class FlashHeterogeneousImpl
+    : public hal::FlashHeterogeneous
+{
+    using hal::FlashHeterogeneous::FlashHeterogeneous;
+
+    void WriteBuffer(infra::ConstByteRange buffer, uint32_t address, infra::Function<void()> onDone) override
+    {}
+
+    void ReadBuffer(infra::ByteRange buffer, uint32_t address, infra::Function<void()> onDone) override
+    {}
+
+    void EraseSectors(uint32_t beginIndex, uint32_t endIndex, infra::Function<void()> onDone) override
+    {}
+};
+
+class FlashHeterogeneousTest
+    : public testing::Test
+{
+public:
+    std::array<uint32_t, 3> sectorSizes{ 3, 4, 8 };
+    FlashHeterogeneousImpl flash{ infra::MakeRange(sectorSizes) };
+};
+
+TEST_F(FlashHeterogeneousTest, number_of_sectors)
+{
+    ASSERT_EQ(3, flash.NumberOfSectors());
```

Comment:
```suggestion
    EXPECT_EQ(3, flash.NumberOfSectors());
```
This and all tests below should really be EXPECTs, not ASSERTs